bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -283,21 +313,56 @@ public void mutate(final Mutation mutation) throws IOException {
   private void offer(Mutation mutation) throws IOException {
     try {
       initializeAsyncMutators();
-      long operationId = heapSizeManager.registerOperationWithHeapSize(mutation.heapSize());
-      MutationOperation operation = new MutationOperation(mutation, operationId, false);
-      if (executorService != null && options.getAsyncMutatorCount() > 0) {
-        mutationsToBeSent.add(operation);
+      Runnable operation = null;
+      if (options.useBulkApi() && (mutation instanceof Put || mutation instanceof Delete)) {
+        // TODO: Do this logic asynchronously. 
+        synchronized (bulkMutationLock) {
```

Comment:
The comment is still true with bulk mutations.  AsyncExecutor.mutateRowsAsync() will make sure that happens.